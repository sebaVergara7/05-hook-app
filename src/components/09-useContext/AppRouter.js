import React from "react";
import {
    BrowserRouter as Router,
    Routes,
    // Switch,
    Route,
    Navigate,
    // Redirect
} from "react-router-dom";

import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />

                <div className='container'>
                    <Routes>
                    {/* <Switch> */}
                        <Route path="/" element={<HomeScreen />} />
                        <Route path="/about" element={<AboutScreen />} />
                        <Route path="/login" element={<LoginScreen />} />
                        <Route path='*' element={<Navigate to='/' />} />
                        {/* <Redirect to='/'/> */}
                    {/* </Switch> */}
                    </Routes>
                </div>

            </div>
        </Router>
    )
}
