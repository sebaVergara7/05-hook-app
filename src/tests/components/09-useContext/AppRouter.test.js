import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en AppRouter.test.js', () => {

    const user = {
        id: 1,
        name: 'Sebasti√°n',
    };

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter /> 
        </UserContext.Provider>
    );

    test('debe mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

});