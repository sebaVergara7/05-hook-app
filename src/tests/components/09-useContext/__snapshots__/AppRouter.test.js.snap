// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter.test.js debe mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "Sebasti√°n"
}
      </pre>
    </div>
  </div>
</div>
`;
